client/src/components/JobList.tsx

```diff
@@ -20,15 +36,50 @@ export default function JobList() {
     loadJobs();
   }, []);
 
+  /**
+   * This block filters the jobs list based on the inputs from FilterForm.tsx
+   * There is a ternary for each input field.
+   * Salary ternary handles the number to string comparison;
+   *   The select options for salary increments by 50000, hence line 55
+   */
```

Comment:
I would extract the filtering to a separate function so it isn't taking up so much space in the component function. You could put it at the bottom of this file, or in a separate file.